File: Solution/MediaBazaar/ProjectClasses/LogicLayer/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectClasses
{
    public class Employee
    {
        //Instance variables
        private int id;
        private string bsn;
        private string firstName;
        private string lastName;
        private Gender gender;
        private string email;
        private string username;
        private DateTime birthDay;
        private string addrStreet;
        private string addrStreetNumber;
        private string addrZipcode;
        private string addrTown;
        private string addrCountry;
        private DateTime firstWorkingDay;
        private string emergencyPhoneNumber;
        private string iban;
        private double hourlyWage;
        private DateTime contractStartDate;
        private ContractType contract;
        private EmployeeType position;

        //Properties
        public int Id
        {
            get { return this.id; }
            private set
            {
                if (value >= 0)
                { this.id = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string BSN
        {
            get { return this.bsn; }
            private set
            {
                if (value != "")
                { this.bsn = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string Street
        {
            get { return this.addrStreet; }
            private set
            {
                if (value != "")
                { this.addrStreet = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string StreetNumber
        {
            get { return this.addrStreetNumber; }
            private set
            {
                if (value != "")
                { this.addrStreetNumber = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string Zipcode
        {
            get { return this.addrZipcode; }
            private set
            {
                if (value != "")
                { this.addrZipcode = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string Town
        {
            get { return this.addrTown; }
            private set
            {
                if (value != "")
                { this.addrTown = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string Country
        {
            get { return this.addrCountry; }
            private set
            {
                if (value != "")
                { this.addrCountry = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string FirstName
        {
            get { return this.firstName; }
            private set
            {
                if (value != "")
                { this.firstName = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string LastName
        {
            get { return this.lastName; }
            private set
            {
                if (value != "")
                { this.lastName = value; }
                else
                    throw new ArgumentException();
            }
        }
        public Gender Gender
        {
            get { return this.gender; }

            set { this.gender = value; }
        }
        public string Username
        {
            get { return this.username; }
            private set
            {
                if (value != "")
                { this.username = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string Birthday
        {
            get { return this.birthDay.ToString("yyyy-MM-dd"); }

        }
        public DateTime BirthDay
        {
            set { this.birthDay = value; }
        }
        public string FirstWorkingDay
        {
            get { return this.firstWorkingDay.ToString("yyyy-MM-dd"); }
        }
        public DateTime FirstWorkingday
        {
            set { this.firstWorkingDay = value; }
        }
        public string Email
        {
            get { return this.email; }
            private set
            {
                if (value != "")
                { this.email = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string PhoneNumber
        {
            get { return this.emergencyPhoneNumber; }
            private set
            {
                if (value != "")
                { this.emergencyPhoneNumber = value; }
                else
                    throw new ArgumentException();
            }
        }
        public EmployeeType Position
        {
            get { return this.position; }

            set { this.position = value; }
        }
        public string ContractStartDate
        {
            get { return this.contractStartDate.ToString("yyyy-MM-dd"); }
        }
        public ContractType Contract
        {
            get { return this.contract; }
            set { this.contract = value; }
        }
        public double HourlyWage
        {
            get { return this.hourlyWage; }

            private set
            {
                if (value > 0)
                { this.hourlyWage = value; }
                else
                    throw new ArgumentException();
            }
        }
        public string Iban
        {
            get { return iban; }

            private set
            {
                if (value != "")
                { this.iban = value; }
                else
                    throw new ArgumentException();
            }
        }

        public string GetNames
        {
            get { return $"ID: {this.id} {this.firstName} {this.lastName}"; }
        }
        public string GetAddress
        {
            get { return $"{addrStreet} {addrStreetNumber}; {addrZipcode}; {addrCountry}, {addrTown}"; }
        }
        public string GetEmployeeNames
        {
            get { return $"{this.firstName} {this.lastName}"; }
        }
        public string GetInfo
        {
            get { return $"ID: {this.Id}      {this.firstName} {this.lastName} \t- {this.Position}\t Contract: {this.Contract}"; }
        }

        //Constructor
        public Employee(int id, string bsn, string firstName, string lastName,
                         Gender gender, string email, string username, DateTime birthDay,
                         string addrStreet, string addrStreetNumber, string addrZipcode,
                         string addrTown, string addrCountry, DateTime firstWorkingDay,
                         string emergencyPhoneNumber, string iban, double hourlyWage,
                         DateTime contractStartDate, ContractType contract, EmployeeType position)
        {
            if (firstWorkingDay<birthDay)
            {
                throw new ArgumentException();
            }
            this.Id = id;
            this.BSN = bsn;
            this.FirstName = firstName;
            this.LastName = lastName;
            this.gender = gender;
            this.Street = addrStreet;
            this.StreetNumber = addrStreetNumber;
            this.Zipcode = addrZipcode;
            this.Town = addrTown;
            this.Country = addrCountry;
            this.Email = email;
            this.Username = username;
            this.birthDay = birthDay;
            this.firstWorkingDay = firstWorkingDay;
            this.PhoneNumber = emergencyPhoneNumber;
            this.Iban = iban;
            this.HourlyWage = hourlyWage;
            this.contractStartDate = contractStartDate;
            this.Contract = contract;
            this.Position = position;
        }

        //Method/s
        public override string ToString()
        {
            return $"Id: {id}\t  {firstName} {lastName} - {position}";
        }
        public int GetMaxContractualHours()
        {
            switch (this.Contract)
            {
                case ContractType.FULLTIME:
                    {
                        return 40;
                    }
                case ContractType.EIGHTYPERCENT:
                    {
                        return 32;
                    }
            }
            return -1;
        }
    }
}

File: website_MediaBazaar/classes/Employee.php
<?php
class Employee
{
        private $id;
        private $bsn;
        private $fname;
        private $lname;
        private $gender;
        private $email;
        private $username;
        private $password;
        private $birthDay;
        private $addrStreet;
        private $addrStreetNumber;
        private $addrZipcode;
        private $addrTown;
        private $addrCountry;
        private $firstWorkingDay;
        private $emergencyPhoneNumber;
        private $iban;
        private $hourlyWage;
        private $contractStartDate;
        private $contract;
        private $position;
    
    public function __construct(int $id, string $bsn, string $firstName, string $lastName, string $gender, string $email, string $username, string $password, string $birthDay, string $addrStreet
    , string $addrStreetNumber, string $addrZipcode, string $addrTown, string $addrCountry, string $firstWorkingDay, string $emergencyPhoneNumber, string $iban, 
    string $hourlyWage, string $contractStartDate, string $contract, string $position)
    {
        $this->id= $id;
        $this->bsn= $bsn;
        $this->fname = $firstName;
        $this->lname = $lastName;
        $this->gender = $gender;
        $this->email = $email;
        $this->username = $username;
        $this->password = $password;
        $this->birthDay = $birthDay;
        $this->addrStreet = $addrStreet;
        $this->addrStreetNumber = $addrStreetNumber;
        $this->addrZipcode = $addrZipcode;
        $this->addrTown = $addrTown;
        $this->addrCountry = $addrCountry;
        $this->firstWorkingDay = $firstWorkingDay;
        $this->emergencyPhoneNumber = $emergencyPhoneNumber;
        $this->iban = $iban;
        $this->hourlyWage = $hourlyWage;
        $this->contractStartDate = $contractStartDate;
        $this->contract = $contract;
        $this->position = $position;
        
    }

    public function getId()
    {
        return $this->id;
    }public function getBsn()
    {
        return $this->bsn;
    }
    public function getFname()
    {
        return $this->fname;
    }
    public function setFname($fname) {
        $this->fname = $fname;
      }
      public function getLname()
    {
        return $this->lname;
    }
    public function setLname($lname) {
        $this->lname = lname;
      }
    public function getUsername()
    {
        return $this->username;
    }
    public function setUsername($username) {
        $this->username = $username;
      }
    public function getPassword()
    {
        return $this->password;
    }
    public function setPassword($password) {
        $this->password = $password;
      }
    public function getEmail()
    {
        return $this->email;
    }
    public function setEmail($email) {
        $this->email = $email;
      }
      public function getPhone()
    {
        return $this->emergencyPhoneNumber;
    }
    public function setPhone($phone) {
        $this->emergencyPhoneNumber = $phone;
      }
      public function getStreet()
    {
        return $this->addrStreet;
    }
    public function setStreet($street)
    {
        $this->addrStreet = $street;
    }
    public function getStreetNumber()
    {
        return $this->addrStreetNumber ;
    }
    public function setStreetNumber($streetNumber)
    {
        $this->addrStreetNumber = $streetNumber;
    }
    public function getZipcode()
    {
        return $this->addrZipcode ;
    }
    public function getTown()
    {
        return $this->addrTown ;
    }
    public function setTown($town)
    {
        $this->addrTown = $town;
    }
    public function getCountry()
    {
        return $this->addrCountry;
    }
    public function setCountry($country)
    {
        $this->addrCountry = $country;
    }
      public function getIban()
    {
        return $this->iban;
    }
    public function setIban($iban) {
        $this->iban = $iban;
      }
      public function getPosition()
    {
        return $this->position;
    }
    public function getGender()
    {
        return $this->gender;
    }
    public function getContract()
    {
        return $this->contract;
    }
   
}
?>